### ISSUE TITLE
enabling round does not give correct result

### ISSUE THREAD

**Issue by tazmaniax** [EvanHahn/HumanizeDuration.js#98]
Hi,

If I use `humanizeDuration(moment().diff(thisDate), { delimiter: ' ', largest: 3 });` then I get something like `21 minutes 16.291 seconds` but if I include the round parameter like `humanizeDuration(moment().diff(thisDate), { delimiter: ' ', largest: 3, round: true });` then I get `0 seconds`

I'm using v3.7. Am I doing something wrong?

thx,
Chris

**Comment by EvanHahn** [comment]
Hmm. Could you provide a sample input (ie, what is the value of `moment().diff(thisDate)`)?

**Comment by tazmaniax** [comment]
thisDate is "2016-04-16T22:13:56+0100" and the duration coming out of moment().diff(thisDate) is 2838550.

**Comment by EvanHahn** [comment]
Hmm, this looks right to me:

```js
humanizeDuration(2838550, { delimiter: ' ', largest: 3 })
// => "47 minutes 18.55 seconds"

humanizeDuration(2838550, { delimiter: ' ', largest: 3, round: true })
// => "47 minutes 19 seconds"
```

Is that the output you're getting?

**Comment by astoellis** [comment]
I'm getting the same result as @tazmaniax . Looks like it's an issue around the rounding issue.
```js
humanizeDuration( 2588982, { largest: 1, round: true } );
```
 Once it enters the `if (options.round)` block, pieces looks as thus:

![image](https://cloud.githubusercontent.com/assets/2126677/14647236/d45b5ec2-062a-11e6-865e-a6e5b20c71da.png)

But it comes out 
![image](https://cloud.githubusercontent.com/assets/2126677/14647453/cb63774a-062b-11e6-820d-3a67d088bf80.png)

and eventually sets them all to 0.

it appears to be looping through and setting the value to `0` ~ [line 373](https://github.com/EvanHahn/HumanizeDuration.js/blob/master/humanize-duration.js#L373):
```js
if ((piece.unitCount % ratioToLargerUnit) === 0 || (options.largest && ((options.largest - 1) < i))) {
    previousPiece.unitCount += piece.unitCount / ratioToLargerUnit
    piece.unitCount = 0
}
```

Might need a bit more digging to figure out exactly what's going on, but I at least narrowed it down to that.

**Comment by EvanHahn** [comment]
Thanks for reporting and digging--I think I may have been testing on an old
version. I'll take a look.

**Comment by EvanHahn** [comment]
Sorry I'm slow to fix this! I've been busy.

**Comment by EvanHahn** [comment]
This has been fixed in `humanize-duration@3.7.1`. Thanks to @brandly for his helpful PR!

**Comment by tazmaniax** [comment]
Thx!
